crates/ibc/src/core/ics03_connection/events.rs

```diff
@@ -32,56 +32,67 @@ pub struct Attributes {
 /// we will be able to remove the `.parse().unwrap()` calls.
 impl From<Attributes> for Vec<Tag> {
     fn from(a: Attributes) -> Self {
-        let mut attributes = vec![];
-        if let Some(conn_id) = a.connection_id {
-            let conn_id = Tag {
-                key: CONN_ID_ATTRIBUTE_KEY.parse().unwrap(),
-                value: conn_id.to_string().parse().unwrap(),
-            };
-            attributes.push(conn_id);
-        }
+        let conn_id = Tag {
+            key: CONN_ID_ATTRIBUTE_KEY.parse().unwrap(),
+            value: a.connection_id.to_string().parse().unwrap(),
+        };
+
         let client_id = Tag {
             key: CLIENT_ID_ATTRIBUTE_KEY.parse().unwrap(),
             value: a.client_id.to_string().parse().unwrap(),
         };
-        attributes.push(client_id);
-        if let Some(conn_id) = a.counterparty_connection_id {
-            let conn_id = Tag {
-                key: COUNTERPARTY_CONN_ID_ATTRIBUTE_KEY.parse().unwrap(),
-                value: conn_id.to_string().parse().unwrap(),
-            };
-            attributes.push(conn_id);
-        }
+
+        let counterparty_conn_id = Tag {
+            key: COUNTERPARTY_CONN_ID_ATTRIBUTE_KEY.parse().unwrap(),
+            value: match a.counterparty_connection_id {
+                Some(counterparty_conn_id) => counterparty_conn_id.to_string().parse().unwrap(),
```

Comment:
Note: opened [this issue](https://github.com/informalsystems/tendermint-rs/issues/1217) in tendermint-rs to eventually clean up the `.parse().unwrap()`.